native/acter/src/api/activities.rs

```diff
@@ -0,0 +1,125 @@
+use std::ops::Deref;
+
+use acter_core::{
+    activities::Activity as CoreActivity,
+    models::{status::membership::MembershipChange as CoreMembershipChange, ActerModel},
+    referencing::IndexKey,
+};
+use futures::{FutureExt, Stream, StreamExt};
+use ruma::{EventId, OwnedEventId, OwnedRoomId, RoomId};
+use tokio::sync::broadcast::Receiver;
+use tokio_stream::wrappers::BroadcastStream;
+
+use super::{Client, RUNTIME};
+
+#[derive(Clone, Debug)]
+pub struct MembershipChange(CoreMembershipChange);
+
+impl MembershipChange {
+    pub fn user_id_str(&self) -> String {
+        self.0.user_id.to_string()
```

Comment:
is this some kind of indexing ? Think we could use ```inner``` or some better access name ?

Reply:
yes, if your struct has unnamed fields (think `struct MytType(String, u32)`) then you access them with the counters. As we only have one inner type, there is no reason to have named fields ...